dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ExclusionFileWriter.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Multimap;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLEventFactory;
+import javax.xml.stream.XMLEventWriter;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.StartElement;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+
+/** Writer for Linkage Checker exclusion files. */
+class ExclusionFileWriter {
+
+  static final XMLEventFactory eventFactory = XMLEventFactory.newInstance();
+
+  private static final QName LINKAGE_CHECKER_FILTER_TAG = QName.valueOf("LinkageCheckerFilter");
+  private static final QName CLASS_TAG = QName.valueOf("Class");
+  private static final QName LINKAGE_ERROR_TAG = QName.valueOf("LinkageError");
+  private static final QName TARGET_TAG = QName.valueOf("Target");
+  private static final QName SOURCE_TAG = QName.valueOf("Source");
+  private static final QName METHOD_TAG = QName.valueOf("Method");
+  private static final QName FIELD_TAG = QName.valueOf("Field");
+
+  /** Writes {@code linkageErrors} as exclusion rules into {@code outputFile}. */
+  static void write(Path outputFile, Multimap<SymbolProblem, ClassFile> linkageErrors)
```

Comment:
I think we need to stop using multimaps. We need a better data type here.

Reply:
How about `LinkageErrors` class that has a field `Multimap<SymbolProblem, ClassFile> errors`? Anyway, such change is outside the scope of this PR.